drivers/volume/portworx/portworx.go

```diff
@@ -2952,6 +2939,107 @@ func (d *portworx) upgradePortworxDaemonset(specGenUrl string) error {
 	return nil
 }
 
+// upgradePortworxStorageCluster upgrades PX StorageCluster based on the given Spec Generator URL
+func (d *portworx) upgradePortworxStorageCluster(specGenUrl string) error {
+	log.InfoD("Upgrading Portworx StorageCluster")
+
+	// Get k8s version
+	k8sVersion, err := k8sCore.GetVersion()
+	if err != nil {
+		return err
+	}
+
+	// Get images from version URL
+	imageList, err := optest.GetImagesFromVersionURL(specGenUrl, k8sVersion.String())
+	if err != nil {
+		return fmt.Errorf("failed to get image list from version URL [%s], Err: %v", specGenUrl, err)
+	}
+
+	// Get PX StorageCluster
+	cluster, err := d.GetDriver()
+	if err != nil {
+		return err
+	}
+
+	// Update PX Image and Env vars if needed
+	updateParamFunc := func(cluster *v1.StorageCluster) *v1.StorageCluster {
+		// Set oci-mon version
+		cluster.Spec.Image = imageList["version"]
+		var envVars []corev1.EnvVar
+		// Add Release Manifest URL incase of edge spec
+		envVars, err := configurePxReleaseManifestEnvVars(cluster.Spec.Env, specGenUrl)
+		if err != nil {
+			return nil
+		}
+		cluster.Spec.Env = envVars
+		return cluster
+	}
+
```

Comment:
Can you please call stats API here. We are doing something like that in upgrade test. Send stats for upgrade-operator. 
`dash.UpdateStats("px-upgrade-stats", "px-enterprise", "upgrade", majorVersion, statsData)`

Reply:
I don't want to add it inside this function, something like that would need to be added in the test itself

Reply:
other upgrade tests will probably need that as well, let's do it as a part of a separate PR